tests/test_performance_optimization_validation.py

```
@@ -0,0 +1,429 @@
+    memory_delta_mb: float
+    success: bool
+    error_message: Optional[str] = None
+    result_hash: Optional[str] = None  # For content validation
+
+
+class PerformanceOptimizationValidator:
+    """Validator for comparing original vs optimized implementations."""
+
+    def __init__(self):
+        self.original_loader = DocumentLoader()
+        self.optimized_loader = OptimizedDocumentLoader()
```

Ruff (0.12.2)

Undefined name `OptimizedDocumentLoader`

(F821)